### ISSUE TITLE
itrax-join: trouble labelling data before join operation

### ISSUE THREAD

**Issue by tombishop1** [tombishop1/itraxR#10]

**Comment by tombishop1** [comment]
I want to create a function that will: append multiple dataframes by row, adding a column with the name of the original dataframe, and then sort the new composite dataframe by the value of a column. So far, I can do this manually quite easily, e.g.:

```
# create some example data that need joining
Pb = c(5, 6, 4, 5, 7)
depth = c(1, 2, 3, 4, 5)
df1 <- data.frame(Pb, depth)
df2 <- df1 - 0.5

# add a column to each with the name of the original dataframe
df1$label <- deparse(substitute(df1))
df2$label <- deparse(substitute(df2))

# join the dataframes together
df <- rbind(df1, df2)

# sort them by depth
df <- df[with( df, order(depth)),]
```

To make this a function, I need to be able to pass an arbitrarily long list of dataframes for labelling and joining. It's the labelling that's the problem, because it won't work in `lapply`. This is what I've got so far, but the labelling doesn't work:

```
# function to label, join and sort data
require(dplyr)
itrax_join=function(list){

# label the data
addlabels=function(x){x$labels<-deparse(substitute(x))}
list <- lapply(list, addlabels)

# join them
df <- bind_rows(list)

# sort them by depth
df <- df[with( df, order(depth) ) , ]

return(df)
}

# example usage
Pb = c(5, 6, 4, 5, 7)
depth = c(1, 2, 3, 4, 5)
df1 <- data.frame(Pb, depth)
df2 <- df1 - 0.5

df3 <- itrax_join(list(df1,df2))
```

Apparently [this is a thing](http://stackoverflow.com/questions/18508790/deparsesubstitutex-in-lapply) to do with `deparse(substitute(x))`, and `lapply`, but I can't for the life of me work around it like[ others have](http://stackoverflow.com/questions/9950144/access-lapply-index-names-inside-fun/9950734#9950734). Can you help, @gcapes ?

**Comment by tombishop1** [comment]
This has been solved by Imo's comment on [my question on Stack Exchange](http://stackoverflow.com/questions/41897684/deparsesubstitutex-in-lapply), and is solved in commit 35231727aed632e4f51a08afaf44e096ed1f7244.
